domains/transfer/Transfer.API/API/Shared/Services/ProjectOriginService.cs

```diff
@@ -0,0 +1,25 @@
+using Grpc.Core;
+using Microsoft.IdentityModel.Tokens;
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System;
+
+namespace API.Shared.Services;
+
+public abstract class ProjectOriginService
```

Comment:
Not really sure if this should be a base class at all and not just a static helper method. But if kept as a base class, the name "Wallet" should be added to indicate that this is for the Wallet.